src/react/components/pages/connections/connectionsPage.tsx

```diff
@@ -120,6 +121,7 @@ export default class ConnectionPage extends React.Component<IConnectionPageProps
     }
 
     private onFormSubmit = async (connection: IConnection) => {
+        connection.connectionType = getStorageType(connection.providerType);
```

Comment:
I'm wondering if there is another way to instantiate the correct storageType versus having a helper method with a switch statement.  Could we instantiate the provider based on the chosed providerType and pull off the storage type directly off of the provider?

Reply:
I'll address this in another PR. I started to refactor some pieces to make this easier, but it is exploding even more